frontend/src/components/Forms/CoffeeChatsForm/CoffeeChatsForm.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useEffect, useState } from 'react';
+import { Form, Dropdown } from 'semantic-ui-react';
+import { Emitters } from '../../../utils';
+import { useSelf } from '../../Common/FirestoreDataProvider';
+import { MembersAPI } from '../../../API/MembersAPI';
+import CoffeeChatsDashboard from './CoffeeChatsDashboard';
+import styles from './CoffeeChats.module.css';
+import CoffeeChatAPI from '../../../API/CoffeeChatAPI';
+
+const CoffeeChatsForm: React.FC = () => {
+  const userInfo = useSelf()!;
+  const [member, setMember] = useState<IdolMember | undefined>(undefined);
+  const [category, setCategory] = useState('');
+  const [membersList, setMembersList] = useState<IdolMember[]>([]);
+  const [approvedChats, setApprovedChats] = useState<CoffeeChat[]>([]);
+  const [pendingChats, setPendingChats] = useState<CoffeeChat[]>([]);
+  const [rejectedChats, setRejectedChats] = useState<CoffeeChat[]>([]);
+  const [isChatLoading, setIsChatLoading] = useState<boolean>(true);
+  const [slackLink, setSlackLink] = useState<string>('');
+  const [bingoBoard, setBingoBoard] = useState<string[][]>([[]]);
+
+  useEffect(() => {
+    MembersAPI.getAllMembers().then((members) => setMembersList(members));
+    CoffeeChatAPI.getCoffeeChatsByUser(userInfo).then((coffeeChats) => {
+      setApprovedChats(coffeeChats.filter((chat) => chat.status === 'approved'));
+      setPendingChats(coffeeChats.filter((chat) => chat.status === 'pending'));
+      setRejectedChats(coffeeChats.filter((chat) => chat.status === 'rejected'));
+      setIsChatLoading(false);
+    });
+    CoffeeChatAPI.getCoffeeChatBingoBoard().then((board) => setBingoBoard(board));
+  }, [userInfo]);
+
+  const coffeeChatExists = (): boolean =>
+    approvedChats.some((chat) => chat.otherMember.netid === member?.netid) ||
+    pendingChats.some((chat) => chat.otherMember.netid === member?.netid);
+
+  const submitCoffeeChat = async () => {
+    if (!member) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Team Member Selected',
+        contentMsg: 'Please select a team member!'
+      });
+    } else if (!slackLink) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Slack Link Entered',
+        contentMsg: 'Please enter a link to your Slack post in #coffee-chats!'
+      });
+    } else if (!category) {
+      Emitters.generalError.emit({
+        headerMsg: 'No Category Selected',
+        contentMsg: 'Please select a category!'
+      });
+    } else if (member.netid === userInfo.netid) {
+      Emitters.generalError.emit({
+        headerMsg: 'Cannot Coffee Chat Yourself',
+        contentMsg: 'Please submit a coffee chat with another member!'
+      });
+    } else if (coffeeChatExists()) {
+      Emitters.generalError.emit({
+        headerMsg: 'Coffee Chat Exists',
```

Comment:
Maybe say something like "Cannot coffee chat the same member for more than one category!"